### PR TITLE
changed: use bezier extraction in evaluation of secondary solution.

### PR DESCRIPTION
**Pull request by akva2**
Faster, and works around issues in the computeBasis method for some models.

@VikingScientist

### CODE REVIEW

File: src/ASM/LR/ASMu3D.C [hunk 1]
```diff
@@ -1982,16 +1990,41 @@ bool ASMu3D::evalSolution (Matrix& sField, const IntegrandBase& integrand,
...
+    basis1.computeBasisValues(gpar[0][i], u, 1);
+    basis2.computeBasisValues(gpar[1][i], v, 1);
+    basis3.computeBasisValues(gpar[2][i], w, 1);
+
+    size_t ib=1; // basis function iterator
+    for(int k=0; k<p3; k++) {
+      for(int j=0; j<p2; j++) {
+        for(int ii=0; ii<p1; ii++, ib++) {
+          B(ib,1) = u[2*ii  ]*v[2*j  ]*w[2*k  ];
+          B(ib,2) = u[2*ii+1]*v[2*j  ]*w[2*k  ]*2.0/dv;
```
Comment (VikingScientist) [inline src/ASM/LR/ASMu3D.C]: I guess this should be divided by du, and not dv

File: src/ASM/LR/ASMu3D.C [hunk 2]
```diff
@@ -1982,16 +1990,41 @@ bool ASMu3D::evalSolution (Matrix& sField, const IntegrandBase& integrand,
...
+    double u[2*p1];
+    double v[2*p2];
+    double w[2*p3];
+    basis1.computeBasisValues(gpar[0][i], u, 1);
+    basis2.computeBasisValues(gpar[1][i], v, 1);
+    basis3.computeBasisValues(gpar[2][i], w, 1);
+
+    size_t ib=1; // basis function iterator
+    for(int k=0; k<p3; k++) {
+      for(int j=0; j<p2; j++) {
```
Comment (VikingScientist) [inline src/ASM/LR/ASMu3D.C]: I would like these to be `jj` and `kk` just for consistency.

**Comment by VikingScientist** [comment]
Tested this and in the case of `du==dv`, then it works as advertised with speedup and with no crash during secondary solution evaluation anymore. Will merge after fixup.

File: src/ASM/LR/ASMu3D.C (hunk 1 above)
Comment (akva2) [reply src/ASM/LR/ASMu3D.C]: nice spotting, bad fingers.

File: src/ASM/LR/ASMu3D.C (hunk 2 above)
Comment (akva2) [reply src/ASM/LR/ASMu3D.C]: done.

**Comment by VikingScientist** [comment]
Jenkins build this please

**Comment by akva2** [comment]
and now you learned the hard way that you have to test things with downstreams.

**Pull request merged**